src/pages/downloads/clients/index.tsx

```diff
@@ -31,14 +32,35 @@ function toggleValue<Type>(array: Type[], value: Type): Type[] {
 }
 
 export default function ClientsPage({ recommended = true }: { recommended?: boolean }) {
+  const history = useHistory();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+
   const [filteredClients, setFilteredClients] = useState<Client[]>([...Clients]);
-  const [filter, setFilter] = useState<ClientFilter>({
+  const [filter, setFilterValue] = useState<ClientFilter>({
     recommended,
-    deviceTypes: [],
-    platforms: []
+    deviceTypes: (searchParams.get('type')?.split(',') ?? []) as DeviceType[],
+    platforms: (searchParams.get('platform')?.split(',') ?? []) as Platform[]
   });
+
   const [isFiltersVisible, setIsFiltersVisible] = useState(false);
 
+  const setFilter = (filter: ClientFilter) => {
+    let search: string[] = [];
+
+    if (filter.deviceTypes.length > 0) {
+      search.push(`type=${filter.deviceTypes.join(',')}`);
+    }
+    if (filter.platforms.length > 0) {
+      search.push(`platform=${filter.platforms.join(',')}`);
+    }
+    history.push({
+      search: '?' + search.join('&')
+    });
```

Comment:
Why not use URLSearchParams to construct the query parameters here? Should be more safe because it applies url encoding etc.